maven-model-builder/src/main/java/org/apache/maven/model/building/DefaultModelBuilder.java

```diff
@@ -1683,13 +1684,20 @@ private void importDependencyManagement(
 
             it.remove();
 
+            // Model v3
             DependencyManagement importMgmt = loadDependencyManagement(model, request, problems, dependency, importIds);
+            if (importMgmt == null) {
+                continue;
+            }
 
-            if (importMgmt != null) {
-                if (importMgmts == null) {
-                    importMgmts = new ArrayList<>();
-                }
-
+            if (request.isLocationTracking()) {
+                // Keep track of why this DependencyManagement was imported.
+                // And map model v3 to model v4 -> importMgmt(v3).getDelegate() returns a v4 object
+                importMgmts.add(
+                        org.apache.maven.api.model.DependencyManagement.newBuilder(importMgmt.getDelegate(), true)
+                                .importedFrom(dependency.getDelegate().getLocation(""))
```

Comment:
importedFrom should not be assigned to DependencyManagement but to each Dependency, isn't it?

Reply:
@hboutemy you're right. It should only map it from model v3 to model v4. We're going to remove the line `.importedFrom(dependency.getDelegate().getLocation(""))`